google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -323,53 +325,47 @@ public static ApiTracerFactory createBigtableTracerFactory(
         .add(MetricsTracerFactory.create(tagger, stats, attributes))
         // Add user configured tracer
         .add(settings.getTracerFactory());
-    Attributes otelAttributes =
-        Attributes.of(
-            PROJECT_ID_KEY, projectId, INSTANCE_ID_KEY, instanceId, APP_PROFILE_KEY, appProfileId);
     BuiltinMetricsTracerFactory builtinMetricsTracerFactory =
-        createBuiltinMetricsTracerFactory(
-            projectId, settings.getMetricsProvider(), otelAttributes, clientContext);
+        openTelemetry != null
+            ? BuiltinMetricsTracerFactory.create(openTelemetry, createBuiltinAttributes(settings))
+            : null;
     if (builtinMetricsTracerFactory != null) {
       tracerFactories.add(builtinMetricsTracerFactory);
     }
     return new CompositeTracerFactory(tracerFactories.build());
   }
 
-  private static BuiltinMetricsTracerFactory createBuiltinMetricsTracerFactory(
-      String projectId,
-      MetricsProvider metricsProvider,
-      Attributes attributes,
-      ClientContext clientContext)
+  @Nullable
+  public static OpenTelemetry getOpenTelemetry(
+      String projectId, MetricsProvider metricsProvider, Credentials defaultCredentials)
```

Comment:
Please mark Credentials as Nullable.

Also is the current behavior is intentional? ie NoCredentialProvider will return null Credentials, but that will be passed down into the metrics client as FixedCredentials (null)